### PR TITLE
feat(plugin-garfish): support microfront-end fast refresh

### PR DESCRIPTION
**Pull request by nyqykk**
## Summary
**Too many users mentioned that when we migrate to Modern.js v2, On Microfront-End, the hmr can not work,even the reload,let alone fast refresh.**

### <samp>🤖 Generated by Copilot at 33abbcb</samp>

Add webpack hook to `garfishPlugin` function to generate unique bundle names for Garfish plugins. This prevents name collisions and enables interoperability between plugins.

### <samp>🤖 Generated by Copilot at 33abbcb</samp>

*  Add webpack hook to generate unique output name for Garfish plugin ([link](https://github.com/web-infra-dev/modern.js/pull/4481/files?diff=unified&w=0#diff-10bba10d09124276d7e6033e0d3df831808249eb1577603c39e2369625917694R154-R158))

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/runtime/plugin-garfish/src/cli/index.ts [hunk 1]
```diff
@@ -151,6 +151,11 @@ export const garfishPlugin = ({
                   .use(webpack.BannerPlugin, [{ banner: 'Micro front-end' }]);
               }
             },
+            webpack: webpackConfig => {
+              if (!webpackConfig.output?.uniqueName) {
```
Comment (chenjiahan) [inline packages/runtime/plugin-garfish/src/cli/index.ts:155]: Can we use `tools.bundlerChain` to set this config? I think Rspack supports uniqueName too.

Comment (chenjiahan) [reply packages/runtime/plugin-garfish/src/cli/index.ts:155]: https://www.rspack.dev/config/output.html#outputuniquename

Comment (nyqykk) [reply packages/runtime/plugin-garfish/src/cli/index.ts]: done，i forgot it, sorry

File: packages/runtime/plugin-garfish/src/cli/index.ts [hunk 2]
```diff
@@ -205,6 +206,9 @@ export const garfishPlugin = ({
                   });
                 }
               }
+              if (!preConfig.output!.uniqueName) {
```
Comment (chenjiahan) [inline packages/runtime/plugin-garfish/src/cli/index.ts:209]: Using `chain.output.get('uniqueName')` is better, `chain.toConfig()` will convert the whole config object and  is slower.

Comment (nyqykk) [reply packages/runtime/plugin-garfish/src/cli/index.ts:209]: ok

**Review by chenjiahan — DISMISSED** [review]

**Comment by chenjiahan** [comment]
CI is failed..

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**